src/inmanta/server/services/resourceservice.py

```diff
@@ -780,3 +781,101 @@ def _get_query_params(
             links["prev"] = url_concat(base_url, previous_params)
         return_value = ReturnValue(response=resource_action_dtos, links=links if links else None)
         return return_value
+
+    @protocol.handle(methods_v2.get_resource_events, env="tid", resource_id="id")
+    async def get_resource_events(
+        self,
+        env: data.Environment,
+        resource_id: ResourceVersionIdStr,
+    ) -> Dict[ResourceIdStr, List[ResourceAction]]:
+        id: Id = Id.parse_id(resource_id)
+        if id.version == 0:
+            raise BadRequest(message=f"Version is missing from argument id: {resource_id}")
+
+        async def get_deploy_actions(
+            resource_id: Id,
+            first_timestamp: Optional[datetime.datetime] = None,
+            last_timestamp: Optional[datetime.datetime] = None,
+        ) -> List[data.ResourceAction]:
+            return await data.ResourceAction.query_resource_actions(
+                environment=env.id,
+                resource_type=resource_id.get_entity_type(),
+                agent=resource_id.get_agent_name(),
+                attribute=resource_id.get_attribute(),
+                attribute_value=resource_id.get_attribute_value(),
+                first_timestamp=first_timestamp,
+                last_timestamp=last_timestamp,
+                action=const.ResourceAction.deploy,
+            )
+
+        current_deploy_start: datetime.datetime
+        last_deploy_start: Optional[datetime.datetime]
+
+        deploy_actions: Iterator[data.ResourceAction] = iter(await get_deploy_actions(id))
+        try:
+            current_deploy: data.ResourceAction = next(deploy_actions)
+            if current_deploy.status != const.ResourceState.deploying:
```

Comment:
I don't know if we need this check, this method could be used outside of the deploy as well, perhaps? 

What I mean is: does this check have a value to us (makes testing easier or makes this code easier)

Reply:
I'd say it's just a guard against incorrect invocation.